#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *


# http://wiki.ros.org/dynamic%20reconfigure/Reviews/7/1/11%20Groups%20API%20review%20API%20Review

def LVL(index):  # type: (int) -> int
    return 1 << index


gen = ParameterGenerator()

gen.add("h_lower", int_t, LVL(0), "h_lower", 0, 0, 180)
gen.add("h_upper", int_t, LVL(0), "h_upper", 180, 0, 180)

gen.add("s_lower", int_t, LVL(0), "s_lower", 0, 0, 255)
gen.add("s_upper", int_t, LVL(0), "s_upper", 20, 0, 255)

gen.add("v_lower", int_t, LVL(0), "v_lower", 0, 0, 255)
gen.add("v_upper", int_t, LVL(0), "v_upper", 255, 0, 255)

gen.add("upper_mask_percent", int_t, LVL(0), "% of upper image which is ignored", 60, 0, 100)

gen.add("erosion_size", int_t, LVL(0), "erosion size", 3, 0, 255)
gen.add("erosion_iter", int_t, LVL(0), "erosion iter", 1, 0, 255)

gen.add("publish_masked", bool_t, LVL(0), "publish masked")

gen.add("blur_size", int_t, LVL(0), "blur size", 2, 0, 255)

exit(gen.generate("igvc_bot", "igvc_bot", "Lanes"))
